src/core/service.c

```diff
@@ -2557,9 +2559,11 @@ static void service_notify_cgroup_empty_event(Unit *u) {
 
         case SERVICE_START:
         case SERVICE_START_POST:
-                /* If we were hoping for the daemon to write its PID file,
-                 * we can give up now. */
-                if (s->pid_file_pathspec) {
+                if (s->type == SERVICE_NOTIFY)
+                        /* No chance of getting a ready notification anymore */
+                        service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_RESOURCES);
```

Comment:
Hmm, "SERVICE_FAILURE_RESOURCES" is our catchall for OS errors (i.e. everything that sets errno)... I think it's the wrong error to use here... I think best would be to add a new error type, maybe SERVICE_FAILURE_PROTOCOL or so, after all a process exiting early and cleanly without sending a notification doesn't follow the protocol...